fluss-server/src/main/java/com/alibaba/fluss/server/zk/ZooKeeperClient.java

```diff
@@ -408,6 +408,13 @@ public Optional<TablePartition> getPartition(TablePath tablePath, String partiti
         return getOrEmpty(path).map(PartitionZNode::decode);
     }
 
+    /** Get partition num of a table in ZK. */
+    public int getPartitionNumber(TablePath tablePath) throws Exception {
+        String path = PartitionsZNode.path(tablePath);
+        Stat stat = zkClient.checkExists().forPath(path);
```

Comment:
Add proection for empty node (no partition created under the path), otherwise, it will throw NPE

```
        if (stat == null) {
            return 0;
        }
```